components/src/widget/widget.ts

```diff
@@ -59,20 +54,26 @@ export class PaymentWidget extends LitElement {
       return
     }
 
-    let walletAddressInfo: WalletAddress
     try {
+      const { apiUrl } = this.configController.config
       const walletAddressUrl = checkHrefFormat(
         toWalletAddressUrl(walletAddress)
       )
 
-      walletAddressInfo = await getWalletAddress(walletAddressUrl)
+      const url = new URL('/wallet', apiUrl)
+      url.searchParams.set('walletAddress', walletAddressUrl)
+
+      const response = await fetch(url)
+      const data = await response.json()
+
+      if (!response.ok) {
+        throw new Error((data as { message?: string }).message)
```

Comment:
We need to get the message from the `error` object so that the widget can display it. Something like:

```suggestion
        throw new Error(data.error?.message)
```
otherwise, the error message from the API is not displayed